src/engine/strat_engine/pool.rs

```diff
@@ -289,9 +289,11 @@ impl StratPool {
     }
 
     /// Expand the physical space allocated to a pool by the value from extend_size()
-    /// Return tne number of Sectors added
-    fn extend_data(&mut self, dm: &DM, current_size: Sectors) -> EngineResult<Sectors> {
-        let extend_size = self.extend_size(current_size);
+    /// Return the number of Sectors added
+    fn extend_data(&mut self, dm: &DM, current_size: DataBlocks) -> EngineResult<Sectors> {
+        // The space backing the pool should be extended by a multiple of the
+        // DATA_BLOCK_SIZE.
+        let extend_size = self.extend_size(current_size.0 * DATA_BLOCK_SIZE);
```

Comment:
DataBlocks doesn't have a ```sectors()``` method because the number of sectors in a DataBlock is not known at the point of definition (unlike MetaBlocks).